R/data-set.R

```diff
@@ -1,7 +1,15 @@
+#' Supported types of the error
+#' @include enum.R
+#' @export
+DataErrorType <- enum(c(
+  "ArithmeticStdDev",
+  "GeometricStdDev"
+))
+
 #' @title DataSet
 #' @docType class
-#' @description  A wrapper around DataRepository exposing convenience methods to use and manipulate dataSets
-#' (typically observed data) containing an X column, a Y column and potentially an Error columns
+#' @description  A class for storage of numerical x- and y-value pairs and optional error for y-values.

```

Comment:
I would prefer to use as few `DataRepository` in user-exposed class as possible.